src/Aspire.Hosting/Dashboard/DashboardLifecycleHook.cs

```diff
@@ -154,4 +189,111 @@ private void ConfigureAspireDashboardResource(IResource dashboardResource)
+                {
+                    // We want to capture the replica of the dashboard resource. This will allow us to get the logs for the dashboard.
+                    // When the resource id is not the same as the resource name, it's the replica.
+                    if (notification.Resource.Name == KnownResourceNames.AspireDashboard && notification.ResourceId != notification.Resource.Name)
+                    {
+                        dashboardResourceId = notification.ResourceId;
+                        break;
+                    }
+                }
+            }
+            catch (OperationCanceledException)
+            {
+            }
+
+            return dashboardResourceId;
+        }
+
+        if (await GetDashboardResourceIdentifier().ConfigureAwait(false) is not string dashboardResourceId)
+        {
+            // The only way this can happen is if DCP is unhealthy, so log a warning.
+            distributedApplicationLogger.LogWarning("Failed to get the dashboard logs.");
+            return;
+        }
+
+        var loggerCache = new Dictionary<string, ILogger>();
+        var defaultDashboardLogger = loggerFactory.CreateLogger("Aspire.Hosting.Dashboard");
+
+        try
+        {
+            // We turned on the JSON formatter for the logger, so we can log the log lines as JSON.
+            await foreach (var batch in resourceLoggerService.WatchAsync(dashboardResourceId).WithCancellation(_shutdownTcs.Token).ConfigureAwait(false))
+            {
+                foreach (var logLine in batch)
+                {
+                    try
+                    {
+                        var logMessage = JsonSerializer.Deserialize(logLine.Content, DashboardLogMessageContext.Default.DashboardLogMessage);
+
+                        if (logMessage is not null)
+                        {
+                            if (!loggerCache.TryGetValue(logMessage.Category, out var logger))
+                            {
+                                // Looks strange to see Aspire.Hosting.Dashboard.Aspire.Dashboard.Category,
+                                // so trim the prefix and append Aspire.Hosting.Why is this important?
+                                // Well there are logs emitting from categories that don't start with Aspire.Dashboard so we want to prefix all logs so that they can be controlled by config.
+                                var categoryTrimmed = logMessage.Category.StartsWith("Aspire.Dashboard.") ?
+                                    logMessage.Category["Aspire.Dashboard.".Length..] : logMessage.Category;
+
+                                loggerCache[logMessage.Category] = logger = loggerFactory.CreateLogger($"Aspire.Hosting.Dashboard.{categoryTrimmed}");
+                            }
+
+                            logger.Log(logMessage.LogLevel, logMessage.EventId, logMessage.Message, null, (s, _) => s);
```

Comment:
Does this lose structured data? Not really important because we're just logging to the console.

Maybe have a TODO to convert the state payload into the right type to pass into logging here.

Reply:
Done